vimms/Controller/misc.py

```diff
@@ -11,140 +8,27 @@ class FixedScansController(Controller):
     A controller which takes a schedule of scans, converts them into tasks in queue
     """
 
-    def __init__(self, ionisation_mode, schedule,
-                 isolation_width=DEFAULT_ISOLATION_WIDTH,
-                 params=None):
+    def __init__(self, schedule, params=None):
         super().__init__(params=params)
-
-        self.ionisation_mode = ionisation_mode
-        self.isolation_width = isolation_width
-        self.schedule = schedule
+        if len(schedule) > 0:
+            self.set_tasks(schedule)
 
     def get_initial_tasks(self):
```

Comment:
How does this interact with `get_initial_scan_params`?

Reply:
ah, saw below, you already fixed it so the first one goes as initial and the rest go as the task lists.